polar-core/src/vm.rs

```diff
@@ -892,6 +923,8 @@ impl PolarVirtualMachine {
         );
 
         match (&left.value(), &right.value()) {
+            (_, Value::Partial(_)) => unreachable!("cannot match against a partial"),
```

Comment:
Should this be an error, not a panic?

Reply:
I believe this is currently an unparseable production, which is why we set it as `unreachable!()`.

Reply:
What if we have:

```
f(x, y) if x matches y;
```

And the user queries with `y` as a partial?

Reply:
The parser treats the RHS of `x matches y` as an instance literal with the class `y`, not as the variable `y`. So I don't believe this case can currently arise, but we made it an error anyway.

Reply:
b85270fb6f69a58fd7bdcfcafe8bfaa946c4708e